sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -102,8 +102,8 @@ protected string GetSessionFilePath()
 
         public override void GlobalTimeoutTearDown()
         {
-            // Only enforce the timeout on playback.
-            if (Mode == RecordedTestMode.Playback)
+            // Only enforce the timeout on playback when debugger is not attached.
+            if (Mode == RecordedTestMode.Playback && !Debugger.IsAttached)
             {
                 base.GlobalTimeoutTearDown();
```

Comment:
This logic should be inside the `base.GlobalTimeoutTearDown`

Reply:
It can be more efficient that way as well